### PR TITLE
Debrief API

### PR DESCRIPTION
**Pull request by webbo89**

### CODE REVIEW

**Comment by webbo89** [comment]
Should now work like this:

`/api/v1/participant-scores/8602885e-de68-4b01-9ff9-cdb99ad493ef/`

```
{
    "id": "8602885e-de68-4b01-9ff9-cdb99ad493ef",
    "scores": [
        {
            "task": "LEGENDSCORE",
            "score": 24,
            "debrief": "YOU ARE A LEGEND"
        }
    ]
}

```

**Comment by DanielGeoffreyHodgkins** [comment]
image

Looks like it works to me

File: participant/views.py [hunk 1]
```diff
@@ -127,3 +127,13 @@ def action(self, request, *args, **kwargs):
...
+
+
+class ParticipantScoreViewSet(viewsets.ModelViewSet):
+    queryset = Participant.objects.all()
+    http_method_names = ('get', 'head', 'options')
+
+    def get_serializer_class(self):
+        if self.action == 'list':
+            return ParticipantScoreSerializer
+        return ParticipantScoreSerializer
```
Comment (fericsepi) [inline participant/views.py]: If you want to use the same serialiser for both list and details you can remove the getter function and just define `serializer` as normal after `queryset`
```
serializer = ParticipantScoreSerializer
```

File: participant/serializer.py [hunk 2]
```diff
@@ -139,3 +150,45 @@ def get_value(action_type, key, actions):
...
+            if('reply_id' in action):
+                tasks = ExerciseEmailReply.objects.get(pk=action['reply_id'])
+                email_task_scores = tasks.scores
+                for email_score in email_task_scores:
+                    score = email_score.value
+                    task = email_score.task
+                    task_score.setdefault(str(task.id), []).append(score)
+
+        response = []
+        for key, score_data in task_score.items():
```
Comment (fericsepi) [inline participant/serializer.py]: I would call `key` `task_id` just to be super clear what the dict key is

File: participant/views.py [hunk 3]
```diff
@@ -132,4 +132,4 @@ def action(self, request, *args, **kwargs):
 class ParticipantScoreViewSet(viewsets.ModelViewSet):
     queryset = Participant.objects.all()
     http_method_names = ('get', 'head', 'options')
-    serializer = ParticipantScoreSerializer
+    serializer_class = ParticipantScoreSerializer
```
Comment (fericsepi) [inline participant/views.py]: 👍

**Pull request merged**